app/models/manageiq/providers/lenovo/physical_infra_manager/parsers/components/network_devices_parser.rb

```diff
@@ -0,0 +1,100 @@
+require_relative 'component_parser'
+
+module ManageIQ::Providers::Lenovo
+  module Parsers
+    class NetworkDeviceParser < ComponentParser
+      class << self
+        #
+        # @param component - some component that has network devices
+        #
+        def parse_network_devices(component)
+          devices = select_network_devices(component)
+          devices.map { |device| parse_device(device) }
+        end
+
+        private
+
+        #
+        # Get all devices with the class `Network controller` in `pciDevices` and `addinCards` field
+        # and remove duplicated devices considering the `pciSubID` field
+        #
+        def select_network_devices(component)
+          devices = []
+          devices << component.try(:pciDevices)&.select { |device| device["class"] == "Network controller" }
+          devices << component.try(:addinCards)&.select { |device| device["class"] == "Network controller" }
```

Comment:
I think that the REST API for 1.4.0 does not have this property on it.  You should be able to use the spec tests/vcrs to validate that this will still work.

Reply:
Also, do you need both the try and the &.  syntax?  My understanding is that .try()  and  ()&  are equivalent behaviours.